docs/architecture/adr-059-evidence-composition-and-lifecycle.md

```diff
@@ -0,0 +1,99 @@
+
+### Hybrid Framework
+
+We allow individual and batch evidence to coexist together, meaning that verification, broadcasting and committing is done depending on the evidence type. A hybrid model effectively undermines any possibility of generalization and Tendermint's evidence module will use switch statements and concrete types instead of interfaces to correctly process the evidence. One could argue that although we could see more evidence on the horizon, this won't be to the extent of justifying generalizing evidence and that each evidence can make it's own unique way through the pipeline and on to being committed to the chain and broadcasted to the application.
+
+A hybrid framework (as well as a batch framework) also surfaces the problem of the interface between Tendermint and the application. Currently, we have one concrete type that captures the relevant information needed to communicate to the application. This concrete type is formed per validator and therefore makes it more difficult to assess cartel (multi-validator) attacks. This begs the question of whether Tendermint should, at the application level, break down the evidence for each of the validators or alter the ABCI data structure to include multiple validators that were involved at that height and thus better capture the full extent of the misbehavior.
+
+Last important note is that we are closest to this implementation than any of the other two.
+
+### Batch Framework
+
+The last approach of this category would be to consider batch only evidence. This works fine with `LightClientAttackEvidence` but would require alterations to `DuplicateVoteEvidence` which would most likely mean that the consensus would send conflicting votes to a buffer in the evidence module which would then wrap all the votes together per height before gossiping them to other nodes and trying to commit it on chain. At a glance this may improve IO and verification speed and perhaps more importantly grouping validators gives the application and Tendermint a better overview of the severity of the attack. However individual evidence has the advantage that it is easy to check if a node already has that evidence meaning we just need to check hashes to know that we've already verified this evidence before. Batching evidence would imply that each node may have a different combination of duplicate votes which may complicate things.
```

Comment:
> Batching evidence would imply that each node may have a different combination of duplicate votes which may complicate things.

exactly. or different alterations of LCEvidence: say one removes 1 vote and sends it. does it count as same evidence? what do you do in such case? or maybe previously it was incomplete (e.g. 20 signatures) and now you send complete (25 signatures) evidence

Reply:
Also, malicious val can vote for 1000000 alternative blocks, which you will end up processing. But if evidence is based on validator, then you can reject other 9999999 evidences and only store 1, for which the validator will be punished!

Reply:
Yes I agree with you. I think the evidence module will need to have some address to evidence look up table to prevent this from happening

Reply:
As to the different combinations thing I think that is tricky